File: src/main/java/ludwig/runtime/Concatenator.java
package ludwig.runtime;

import java.util.function.Consumer;

public class Concatenator implements Consumer<Object> {
    private final StringBuilder builder = new StringBuilder();

    @Override
    public void accept(Object o) {
        if (o instanceof String) {
            builder.append((String) o);
        } else if (o instanceof Double) {
            builder.append(((Double) o).doubleValue());
        } else if (o instanceof Long) {
            builder.append(((Long) o).longValue());
        } else {
            builder.append(o);
        }
    }

    @Override
    public String toString() {
        return builder.toString();
    }
}

File: src/main/java/ludwig/runtime/Concatenator.kt
package ludwig.runtime

import java.util.function.Consumer

class Concatenator : Consumer<Any> {
    private val builder = StringBuilder()

    override fun accept(o: Any) {
        if (o is String) {
            builder.append(o)
        } else if (o is Double) {
            builder.append(o.toDouble())
        } else if (o is Long) {
            builder.append(o.toLong())
        } else {
            builder.append(o)
        }
    }

    override fun toString(): String {
        return builder.toString()
    }
}
